src/main/java/org/fogbowcloud/saps/engine/core/dispatcher/email/TasksEmailBuilder.java

```diff
@@ -22,39 +22,24 @@
     private static final Logger LOGGER = Logger.getLogger(TasksEmailBuilder.class);
     private static final String EMAIL_TITLE = "[SAPS] Results of selected tasks";
 
-    private final class Task {
-        private static final String ID = "taskId";
-        private static final String IMAGE_REGION = "task_image_region";
-        private static final String IMAGE_COLLECTION_NAME = "task_image_collection_name";
-        private static final String IMAGE_DATE = "image_date";
-        private static final String ACCESS_LINKS = "access_links";
-
-        private final class AccessLink {
-            private static final String NAME = "name";
-            private static final String URL = "url";
-        }
-    }
-
-    private final PermanentStorage permanentStorage;
-    private final DatabaseApplication application;
-    private final List<String> tasksId;
+    private final List<TaskEmail> tasksEmail;
+    private final Gson jsonUtil;
     private final String userEmail;
     private final String noReplyEmail;
     private final String noReplyPass;
 
-    public TasksEmailBuilder(DatabaseApplication databaseApplication, PermanentStorage permanentStorage,
-                             Properties properties, String userEmail, List<String> tasksId) throws InvalidPropertyException {
+    public TasksEmailBuilder(Properties properties, String userEmail, List<TaskEmail> tasksEmail)
+            throws InvalidPropertyException {
```

Comment:
Well, that class do not have the functions of a Builder, but of a Sender. If you want to make it a Builder let it build all the email. Otherwise, rename him.

Reply:
Done